.github/workflows/release.yml

```diff
@@ -0,0 +1,388 @@
+            git push origin $BRANCH_NAME
+
+            git tag $GH_TAG
+            git push origin $GH_TAG
+          fi
+
+
+      - name: Operator - install operator-sdk
+        run: |
+          mkdir -p $TOOLS_PATH
+          echo $TOOLS_PATH >> $GITHUB_PATH
+
+          export OPERATOR_SDK_DL_URL=https://github.com/operator-framework/operator-sdk/releases/download/v1.25.2
+          curl -LO ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH}
+
+          gpg --keyserver keyserver.ubuntu.com --recv-keys 052996E2A20B5C7E
+
+          curl -LO ${OPERATOR_SDK_DL_URL}/checksums.txt
+          curl -LO ${OPERATOR_SDK_DL_URL}/checksums.txt.asc
+          gpg -u "Operator SDK (release) <cncf-operator-sdk@cncf.io>" --verify checksums.txt.asc
+
+          grep operator-sdk_${OS}_${ARCH} checksums.txt | sha256sum -c -
+
+          chmod +x operator-sdk_${OS}_${ARCH}
+          mv operator-sdk_${OS}_${ARCH} $TOOLS_PATH/operator-sdk
+
+      - name: Operator - build and bundle
+        run: |
+          cd everest-operator
+          make build manifests bundle
+
+
+      - name: Operator - setup Docker meta for everest-operator
+        id: operator_meta
+        uses: docker/metadata-action@v4
+        # docker/metadata-action action looks more elegant when being triggered by a GH tag,
+        # however this workflow can't be triggered by a GH tag since there are some changes need to be done
+        # in the codebase prior putting the tag, so the action uses the raw tags
+        with:
+          images: |
+            percona/everest-operator,enable=${{ env.IS_RC == 0 }}
+            perconalab/everest-operator
+          tags: |
+            type=raw,value=${{ env.VERSION }}
+            type=raw,value=latest
+            type=raw,value=${{ env.FLOATING_TAG }},enable=${{ env.IS_RC == 0 }}
+
+      - name: Operator - setup Docker meta for everest-operator-bundle
+        id: bundle_meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            percona/everest-operator-bundle,enable=${{ env.IS_RC == 0 }}
+            perconalab/everest-operator-bundle
+          tags: |
+            type=raw,value=${{ env.VERSION }}
+            type=raw,value=${{ env.FLOATING_TAG }},enable=${{ env.IS_RC == 0 }}
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Operator - build and push operator everest-operator image
+        uses: docker/build-push-action@v3
+        with:
+          context: everest-operator
+          push: true
+          tags: ${{ steps.operator_meta.outputs.tags }}
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@0.12.0
+        with:
+          image-ref: ${{ steps.operator_meta.outputs.tags }}
+          format: 'table'
+          exit-code: '1'
+          severity: 'CRITICAL,HIGH'
```

Comment:
Shouldn't we do this before pushing the image?

Reply:
The tags that this check scans (`operator_meta` tags - the `everest-operator` images) are already pushed at the previous step.  
The next step below pushes the `bundle_meta` tags (the `everest-operator-bundle` images) and another Trivy scanner runs afterwards to check them.

Reply:
I know that they are already pushed at the previous step, that was my point. If we find a vulnerability shouldn't we block the image from being pushed?

Reply:
Ran some tests, it's possible first to build an image, then scan it and then push. Updated the flow for each of 4 docker images the RC build. 
Also realized that Trivy can't scan a list of tags, it needs only a single one as an input. So added an additional step that defines the first image tag from the list we've just built (it doesn't really matter which one since the build itself is the same) for the Trivy scan step.